bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/client/RpcClient.java

```diff
@@ -83,14 +85,35 @@ public RpcClient(HomematicConfig config) {
     /**
      * Register a callback for the specified interface where the Homematic gateway can send its events.
      */
-    public void init(HmInterface hmInterface, String clientId) throws IOException {
+    public void init(HmInterface hmInterface) throws IOException {
         RpcRequest<T> request = createRpcRequest("init");
         request.addArg(getRpcCallbackUrl());
-        request.addArg(clientId);
+        request.addArg(thisUID);
         if (config.getGatewayInfo().isHomegear()) {
             request.addArg(Integer.valueOf(0x22));
         }
-        sendMessage(config.getRpcPort(hmInterface), request);
+        int maxRetryAttempts = config.getCallbackRegistrationRetries();
+        int waitTime = config.getRetryWaitTime() * 1000;
+        for (int attempt = 1; attempt <= maxRetryAttempts; attempt++) {
+            try {
+                logger.debug("Register callback for interface {}, attempt {}/{}", hmInterface.getName(), attempt,
+                        maxRetryAttempts);
+                sendMessage(config.getRpcPort(hmInterface), request);
+                break;
+            } catch (IOException e) {
+                if (attempt == maxRetryAttempts) {
+                    logger.warn(
+                            "All attempts to register a callback for interface '{}' failed, last exception message: {}",
+                            hmInterface.getName(), e.getMessage());
+                    throw (e);
+                }
+                try {
+                    Thread.sleep(waitTime);
+                } catch (InterruptedException e1) {
+                    // ignore
```

Comment:
The `InterruptedException` is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done. You can achieve this by not catching the exception here but throw the InterruptedException and catch it at the highest level you can.

Reply:
If I replace the `sleep`, the catch is no longer needed

Reply:
@fwolter changed it